File: backend/pkg/models/message.go
package models

import "time"

type Message struct {
	ID          string    `json:"id"`
	SenderID    string    `json:"sender_id"`
	RecipientID *string   `json:"recipient_id,omitempty"`
	GroupID     *string   `json:"group_id,omitempty"`
	Content     string    `json:"content"`
	IsEmojiOnly bool      `json:"is_emoji_only"`
	CreatedAt   time.Time `json:"created_at"`
}

File: frontend/src/types/message.ts
export interface Message {
    id?: string;
    sender_id: string;
    recipient_id?: string;
    group_id?: string;
    content: string;
    is_emoji_only: boolean;
    created_at?: string;
    channel: string;
    event: string;
    data?: Record<string, any>;
  }
  
export interface ChatOutboundMessage {
  channel: "chat";
  event: "private_message" | "group_message";
  to: string;
  content: string;
  data?: Record<string, any>;
}

